src/app/Main/Body/QueryPage/Drawer/Displayed_Features/DisplFeatures-Selector.component.tsx

```diff
@@ -1,13 +1,26 @@
 import React from "react";
 import { ValueType } from "react-select";
+import { useTranslation } from "react-i18next";
+import { TFunction } from "i18next";
 import _ from "lodash";
 import { SelectorComponent } from "../../../../../Shared/Selector.component";
 import { TableType } from "../../../../../Shared/Context/TableContext";
-import {
-    FilterType,
-} from "../../../../../Shared/Model/Filter.model";
+import { FilterType } from "../../../../../Shared/Model/Filter.model";
 import { generateFeatureList } from "../../../../../Core/generateFeatureList.service";
 
+function generateTranslatedSelectorObject(
+    selectorArray: string[],
+    translateFunction: TFunction
+): { value: string; label: string }[] {
+    const selectorLabelObject: {
+        value: string;
+        label: string;
+    }[] = selectorArray.map((selectorElement: string) => {
+        const label: string = translateFunction(`Filters.${selectorElement}`);
+        return { value: selectorElement, label };
+    });
+    return selectorLabelObject;
+}

```

Comment:
can be simplified: return selectorArray.map(...)

Reply:
done .